### PR TITLE
Med rework

### PR DESCRIPTION
**Pull request by goose4429**
## About The Pull Request

Old balance problem pills vs injectors. This PR is a take on making injectors more viable and hampering the "every unga runs with 1000u of chems" meta.

1. Buffs injectors to 30u, makes default inject volume 10u.
2. Doubles injector quantity in MarineMed.
3. Adds more (and fixes old) injectors, corresponding to all pill bottles vended from old MarineMed.
4. Most pill bottles moved to NanoTrasenMedPlus. MarineMed has Trico, Bica, Kelo, Trama and Dylo.
5. TBD: pill use delay on self.

## Why It's Good For The Game

1. Pill bottles were too much better than other med containers. Equalizes distribution of container types.
2. Unga will still be able to carry any kind of chem he could before, but now in less volume.
3. Medics are more important now, and marine will actually ask for help when injured. Not just funny guy with evac.
4. It's still possible to throw pills out of med vendor by docs, hack med vendor, etc.
5. Chem lab is still available as well.
6. Tivi approved idea so obviously 10/10.

## Changelog
:cl:
add: More injectors
balance: Injectors are bigger
balance: Moved some pill bottles to med vendor
balance: These pill bottles removed from MarineMed
balance: small time delay before pill feed to self
/:cl:

### CODE REVIEW

**Comment by EvilMelonTheSecond** [comment]
Hacking has been largely removed from the game, I would recommend not making any new additions relying on it since it might get removed if it gets too involved in gameplay.

**Comment by goose4429** [comment]
> Hacking has been largely removed from the game, I would recommend not making any new additions relying on it since it might get removed if it gets too involved in gameplay.

Hacking is in the game, it works, and I don't rely on it.

> Also this fixes nothing, pills are still superior in every way and I expect nobody to change their loadout with this PR

Unga carries QC, Alkysine and Hypervene pills around. So they'll change, at least some of them. But probably not a lot, yes. PR shouldn't break the meta too much, that's not my goal. Just less pills and more injectors, aye?

**Comment by Lumipharon** [comment]
I forsee the great chem break in, as ungas empty their infinite injectors to make their own pill bottles.

Reminder chem should be ID locked.

**Comment by Arctrooper8** [comment]
### Overall this sounds like a good change but I have two major suggestions/worries:

**1. Ensure that medics have some guaranteed way to get the pills they need (stuff like alky, imi, peri, qc, etc) maybe edit the belt starting content?
2. Fully explore the autoinjector aspect, replace all pill packets with the new autoinjectors and really emphasize their role as the untrained medical tool.**

Some more minor grips include:
Hacking should never be a part of gameplay and is only a leftover from the ancient baystation code. The only reason it's still in is because it's hard coded into vendor code.

And how will you ensure medics get the the now increasingly limited pill bottles? It's already difficult to late join as a medic and discover that half of the specialized pill bottles you need don't exist, and not everyone wants to do chem. 

Also simply moving them to a vendor won't prevent ungas from getting them even with hacking removed. I believe (but I'm not sure) that loadout vendors can pull items from vendors you can't even access.

The pill delay makes sense but I'd be afraid of it feeling awful in practice.

**Comment by Arctrooper8** [comment]
> I forsee the great chem break in, as ungas empty their infinite injectors to make their own pill bottles.
> 
> Reminder chem should be ID locked.

The ID itself could be ID locked but that leads down a worrying path.

**Comment by Lumipharon** [comment]
> The Machine itself could be ID locked but that might be too gamey (thoughts of ID locked pill bottles come to mind).

ID locked machines already exist (req console etc) and actually make sense, unlike CM unga proof pill bottles.
But if you try and remove access to pill bottles, marines will simply do what they have to do to get access again if it's within their power.

So chem is the main one, then they'll also likely loot the shit out of med storage as well.

**Comment by Derrick2Electricboogaloo** [comment]
I know for sure when I eat a pill in real life it does not take me .7 seconds to put it in my mouth

**Comment by Pariah919** [comment]
Pills having self use times seems silly, I guess it gives syringes and bottles a gimmick over pills, though.

**Comment by Derrick2Electricboogaloo** [comment]
> > I forsee the great chem break in, as ungas empty their infinite injectors to make their own pill bottles.
> > Reminder chem should be ID locked.
> 
> The Machine itself could be ID locked but that might be too gamey (thoughts of ID locked pill bottles come to mind).

This ain't CM, keep that over there.

**Comment by Losenis** [comment]
One thing that injectors could use to give them additional QoL vs pill bottles is for them to come in injector cases of 3 like what the normal lifesaver belt has. Then they'd also need some way to store the spent ones on the first slot and full ones to come last, so that quickdrawing them would take out the fresh one than the spent one. Otherwise, there's just no competition.

Or have the injector pouch fit more. If I remember well, the combat injector pouch's total potential reagent storage doesn't even surpass that of one pill bottle.

**Comment by Arctrooper8** [comment]
> > The Machine itself could be ID locked but that might be too gamey (thoughts of ID locked pill bottles come to mind).
> 
> ID locked machines already exist (req console etc) and actually make sense, unlike CM unga proof pill bottles. But if you try and remove access to pill bottles, marines will simply do what they have to do to get access again if it's within their power.
> 
> So chem is the main one, then they'll also likely loot the shit out of med storage as well.

A skill check would make more sense thinking about it. Since you can't hand out skill improvements like you can edit IDs.

**Comment by goose4429** [comment]
> ### Overall this sounds like a good change but I have two major suggestions/worries:
> **1. Ensure that medics have some guaranteed way to get the pills they need (stuff like alky, imi, peri, qc, etc) maybe edit the belt starting content? 2. Fully explore the autoinjector aspect, replace all pill packets with the new autoinjectors and really emphasize their role as the untrained medical tool.**
> 
> Some more minor grips include: Hacking should never be a part of gameplay and is only a leftover from the ancient baystation code. The only reason it's still in is because it's hard coded into vendor code.
> 
> And how will you ensure medics get the the now increasingly limited pill bottles? It's already difficult to late join as a medic and discover that half of the specialized pill bottles you need don't exist, and not everyone wants to do chem.
> 
> Also simply moving them to a vendor won't prevent ungas from getting them even with hacking removed. I believe (but I'm not sure) that loadout vendors can pull items from vendors you can't even access.
> 
> The pill delay makes sense but I'd be afraid of it feeling awful in practice.

Thanks for a good feedback!

TL;DR 
- medics will have less problems with specialized meds with thus PR anyway.
- hacking is just a thing I can't control, I don't count on it
- pill packets to injector is kinda strange TBH

0. All belts/pouches have items in them by default. These items are not taken from vendors, but are just spawned.

1. Medics will have easier access to pills than before, so no problems. Unga doesn't take QC, because it's in NT vendor. You had it before in the MM, now it's just moved so no marines with half of medbay loadout. Ina is in MM, so maybe problematic to get bonus bottle roundstart - but you always have reagent bottles you can quickly mix into the stuff you want. If you want extra, of course. And probably you will have some in MM still, I haven't changed the numbers on amount of pill bottles and limited access by non-medics to them. 

2. Remove pill packets? Uh... not sure. We have it as half-pill-bottle, maybe thats how it should be. A thing to consider in future PRs, I'm nor sure it should be included in this one.

**Comment by goose4429** [comment]
Pill use time - yeah, strange thing. Will make pills taken out with small delay instead, and only for pill bottles.

**Comment by breadhunt** [comment]
CL says you're upping autoinjector size but I don't think that's been changed yet. If they're staying tiny sized, please consider:
- In satchel/backpack/etc. storage, 3 tiny items = 1 small item
- In everything else (e.g webbings), 1 tiny item = 1 small item 
- This could mean it becomes borderline useless to store medical supplies anywhere except pouches/satchel. Which already happens, it's just that you can use pills instead to ignore the issue.
- Loadout flexibility good

This should be pretty easy to avoid, just make a free, small-sized, disposable container that stores 2-3 autoinjectors.

This addresses the issue without adding storage creep and makes managing injectors easier in general. 
If lag is a concern they can always just qdel themselves when the last injector is taken out, or something

**Comment by goose4429** [comment]
> CL says you're upping autoinjector size but I don't think that's been changed yet. If they're staying tiny sized, please consider:

I think there is a confusion created by my wording. I've doubled the injector internal volume, with no planned changes on item size.

**Comment by ilikesaying** [comment]
Volume is the space inside the object, size is the space the object takes up.

File: code/game/objects/items/reagent_containers/autoinjectors.dm [hunk 1]
```diff
@@ -125,29 +122,51 @@
...
 
 /obj/item/reagent_containers/hypospray/autoinjector/inaprovaline
 	name = "inaprovaline autoinjector"
-	desc = "An auto-injector loaded with 15 units of inaprovaline, an emergency stabilization medicine for patients in critical condition."
+	desc = "An auto-injector loaded with 2 doses of inaprovaline, an emergency stabilization medicine for patients in critical condition."
+	icon_state = "autoinjector-9"
+	amount_per_transfer_from_this = 15
+	list_reagents = list(/datum/reagent/medicine/inaprovaline = 30)
+
+/obj/item/reagent_containers/hypospray/autoinjector/dexalin
```
Comment (novaepee) [inline code/game/objects/items/reagent_containers/autoinjectors.dm:140]: Observe how the autoinjector's path is /obj/item/reagent_containers/hypospray/autoinjector/**dexalin**, but the autoinjector name is **inaprovaline**.

File: code/game/objects/items/reagent_containers/autoinjectors.dm [hunk 2]
```diff
@@ -125,29 +122,51 @@
...
+	desc = "An auto-injector loaded with 3 doses of alkysine, long-living medicine for fixing brain damage."
+	icon_state = "autoinjector-1"
+	volume = 15
+	amount_per_transfer_from_this = 5
+	list_reagents = list(/datum/reagent/medicine/alkysine = 15)
+
+/obj/item/reagent_containers/hypospray/autoinjector/imidazoline
+	name = "imidazoline autoinjector"
+	desc = "An auto-injector loaded with 3 doses of imidazoline, medicine for fixing eyesight."
+	icon_state = "autoinjector-1"
```
Comment (novaepee) [inline code/game/objects/items/reagent_containers/autoinjectors.dm:167]: Forgot to include the volume in /obj/item/reagent_containers/hypospray/autoinjector/imidazoline

File: code/game/objects/items/reagent_containers/autoinjectors.dm [hunk 3]
```diff
@@ -171,6 +190,7 @@
 /obj/item/reagent_containers/hypospray/autoinjector/hyperzine/expired
```
Comment (novaepee) [inline code/game/objects/items/reagent_containers/autoinjectors.dm:190]: Likely not used, best to remove this.

File: code/game/objects/items/reagent_containers/autoinjectors.dm [hunk 4]
```diff
@@ -38,13 +38,13 @@
...
-	list_reagents = list(/datum/reagent/medicine/tricordrazine = 15)
+	list_reagents = list(/datum/reagent/medicine/tricordrazine = 30)
 
 /obj/item/reagent_containers/hypospray/autoinjector/combat
 	name = "combat autoinjector"
-	desc = "An autoinjector loaded with two doses of healing and painkilling chemicals. Intended for use in active combat."
+	desc = "An autoinjector loaded with 2 doses of healing and painkilling chemicals. Intended for use in active combat."
 	icon_state = "autoinjector-4"
+	amount_per_transfer_from_this = 15
 	volume = 30
```
Comment (novaepee) [inline code/game/objects/items/reagent_containers/autoinjectors.dm:48]: Can delete this; it already inherit `volume = 30` from `/obj/item/reagent_containers/hypospray/autoinjector`

File: code/game/objects/items/reagent_containers/autoinjectors.dm [hunk 5]
```diff
@@ -55,8 +55,9 @@
 
 /obj/item/reagent_containers/hypospray/autoinjector/combat_advanced
 	name = "Advanced combat autoinjector"
-	desc = "An autoinjector loaded with two doses of advanced healing and painkilling chemicals. Intended for use in active combat."
+	desc = "An autoinjector loaded with 2 doses of advanced healing and painkilling chemicals. Intended for use in active combat."
 	icon_state = "autoinjector-7"
+	amount_per_transfer_from_this = 15
 	volume = 30
```
Comment (novaepee) [inline code/game/objects/items/reagent_containers/autoinjectors.dm:61]: Ditto

File: code/game/objects/items/reagent_containers/autoinjectors.dm [hunk 6]
```diff
@@ -125,29 +122,51 @@
...
+	amount_per_transfer_from_this = 5
+	list_reagents = list(/datum/reagent/medicine/alkysine = 15)
+
+/obj/item/reagent_containers/hypospray/autoinjector/imidazoline
+	name = "imidazoline autoinjector"
+	desc = "An auto-injector loaded with 3 doses of imidazoline, medicine for fixing eyesight."
+	icon_state = "autoinjector-1"
+	list_reagents = list(/datum/reagent/medicine/imidazoline = 30)
+
 /obj/item/reagent_containers/hypospray/autoinjector/hypervene
```
Comment (novaepee) [inline code/game/objects/items/reagent_containers/autoinjectors.dm:170]: Reason for no change to hypervene autoinjector?

Comment (Otterwatch) [reply code/game/objects/items/reagent_containers/autoinjectors.dm:170]: I believe hypervene autoinjector is meant to be a lower unit amount since you only need about 1-2 units at any time. Makes it need use the autoinjector mechanics instead of being a free 30 units of hypervene in a space you will never use for anything else I guess.

Comment (goose4429) [reply code/game/objects/items/reagent_containers/autoinjectors.dm:170]: > Reason for no change to hypervene autoinjector?

3u is OD for hypervene. Pills also have 3u inside. 3 does are in line with others so looks fine to me.
Thanks for code review above!

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Can you please explicitly state what you changed in the CL

**Comment by goose4429** [comment]
All changes are indicated in CL now. Hope I didn't forget anything.
Don't have much space to give the list of items in CL. MarineMed has x6 pill bottle of Trico, Bica, Kelo, Trama, Dylo. No more pill bottles. Pill bottles which were in MM before are now moved to NTMedPlus (med vendor). MM has contents as on the screenshot above.

File: code/game/objects/items/storage/firstaid.dm [hunk 7]
```diff
@@ -368,9 +376,11 @@
 	storage_slots = null
 	use_sound = 'sound/items/pillbottle.ogg'
 	max_storage_space = 16
+	access_delay = 0.5 SECONDS
 	greyscale_config = /datum/greyscale_config/pillbottle
 	greyscale_colors = "#d9cd07#f2cdbb" //default colors
 	var/pill_type_to_fill //type of pill to use to fill in the bottle in New()
+	var/check_access_delay = FALSE
```
Comment (BraveMole) [inline code/game/objects/items/storage/firstaid.dm:383]: autodoc (see dev general pins if you don't know what it is)

**Review by BraveMole — APPROVED** [review]
Changes look sensible, I'm up for tm.

There is probably more work to do on medical, so that "Medics are more important now, and marine will actually ask for help when injured.", but you look to be going in the right direction and you've shown you are ready to put the work and take the feedback.

**Comment by bowlofcereal** [comment]
In practice this has not really changed anything in game. People still use pills. If anything this incentivizes premedication due to the delay. Its not a terrible idea, don't get me wrong, it just does not work out in practice with it test merged.

If we want to give people a reason to use injectors, perhaps make an item that corpsmen can use that will refill injectors on the field with a limited power use? Maybe even like an engineers welderpack, but for corpsmen? I know this is some serious ideas guy'ing but I don't really know where else to put this. I'm not against the idea of changing medical to be less pill-based but this isn't solving the issue unfortunately. It's something that could work if it had crutch PRs built around it to further incentivize injectors over pills.

**Comment by Xerkie** [comment]
this changes almost nothing about balance and just makes things more tedious

hell, it incentivizes pre-medding even MORE so that you already have all the chems inside you with the pill delay, since you don't want to be caught popping pills mid-combat

marines also still break into chem and press every single pill they'd ever need anyways, as they have done for the past several months continuously, which no medic/doctor tries to stop because it's a TDM and they want their team to win, only now marines spend a few minutes longer inside chem to get their vitals

it doesn't go nearly far enough to actually disincentivize pill usage in any capacity, marines still play exactly the same and still get all the pills they need, only now they spend a bit longer in chem and premed even harder so they never get caught off-guard

**Comment by Losenis** [comment]
There's no incentive to use anything other than pills. The closest thing I've seen to actually buffing non-pills was bandages and the sort chain-applying.

You can nerf pills to the absolute ground and marines will still use them if the alternatives are still worse.

I'd still argue there should be free infinite injector cases in the vendors to increase the potential per-slot quantity of injectors, since they cap out at 30u while pill bottles still have up to 240u. I can see injector cases of peridaxon and QC being very useful given you can set them to 1u and microdose (selfsplinting is so absolute dogshit that going without peridaxon is almost suicide given you can get organ damage without bonefracs too). Microdosing is something pills can't do, so it'd be best to encourage it with injectors to have some competition with them; even though pill bottles have 240u of reagents, they're 16 uses at most, while injector cases in the extreme case of 1u can have 90 uses making them very convenient when there's a defiler around that'll purge your shit regardless. Granted, that's an extreme case, I imagine a more practical if still extreme end would be 3u that'd land at around the same usage a pill bottle gives, so injector cases would probably need to be buffed in the amount they can carry. Internally it'd kind of balance itself because you still need to juggle around injectors in the cases that are stored inside other storage areas making it difficult mid-fight.

**Comment by goose4429** [comment]
Thanks for the feedback everybody.

Once again, I'm not trying to get rid of pills - I'm just trying to get alternatives better. Getting rid of pills requires quite a bit more work, and separate balance design which is in plans by Tivi and Mole. So far this one is just a "step towards the right direction". Let's see what will actually happen when wider audience will see the changes, and how many of them will switch to injectors. Basic pill bottles are still in MM, as you can see, just not the specialized ones.

Injectors have been buffed, so instead of 30u per slot it is now actually 60u (1 bottle = 2 injectors). Still worse than 240u by a lot, but at least viable (also note that per use it's 10u by default, so you can spread them more evenly). And combat injectors are very popular, maybe now that they are unlimited people will use them even more.

Focus is on making more specialized chems less abundant for ungas. If you want to have ImiAlky - sure, just take the injectors. But you won't be able to have it on you 24/7 or premed with it since the volume is limited. On the other hand, this will allow to have ImiAlky in one slot instead of 2 slots of bottles. Same with QC and Peri. I am positive that people will start to use them more.

Chemistry lab is always an option for marines. Not _all_ the marines take this option, so to at least _some extent_ this will change something. I do not expect radical changes anyway, just that corpsmen become more important, especially when marines run out of special meds.

**Comment by Losenis** [comment]
I'd be wary about making the specialized chems less abundant, since they tend to be the ones that have little to no alternative. For example there's no real alternative to peridaxon or imialky other than alt tabbing out of the game after a medevac. It may come as a surprise to some c*ders but soft-kicking a player out of the game is neither fun nor balanced overall.

Another issue injector might have is that sometimes you need a sledgehammer approach to what you're struggling with since the average unga will fumble the healthscan unless they have valk or vali, so they can't really go "Oh I have 4u of neurotoxin killing me, I can inject 1u of hypervene", they'll probably be on 0 stam getting knocked down on the ground with a rainbow screen not being able to tell if they have 4u or 40u of neurotox in them. Same with damage, you might have 20 brute damage or skirting 120. Because that tends to happen way more often than not, injectors would need additional incentives to overcome that.

Still a good few starting steps thogh, other than the pill nerfs.

**Comment by FosterGrime-Heymynameslouie** [comment]
this pr is good because it allows me to mass drop combat injectors because it now makes it infinite
based pr speedmehrg instantly

**Comment by Lumipharon** [comment]
I assume issue #10217 is related to this PR.

**Comment by goose4429** [comment]
> I assume issue #10217 is related to this PR.

Correct. I didn't close my last PR with injectors, and it got merged before I fixed stuff. In this PR, this and some other issues are fixed.

**Comment by Xerkie** [comment]
with the updated PR, it's hardly really noticeable since every unga will just hack/deposit fifty autoinjectors into a beaker and then press them to get their beloved pills for essentials like Peridaxon anyways, but i do like the autoinjector buffs a lot and i greatly enjoy not being forced to spam-press my own pills whenever i latejoin due to every single unga in existence having pre-vended them

pills are still 100% the superior option every single time with no real reason to have autoinjectors over them, but autoinjectors are at least not complete shit when they gotta be used now (ie with inaprov for the crit reaction) + the changes to how many pills are stocked in the machines is very nice

good pr

**Comment by zzurge** [comment]
I remember making a github account some time ago to protest against the removal of smartgunner from the game,since then there have been changes I have agreed with and disagreed with,but few I have disagreed with enough to say something,this is one of those times I'm saying something,terribly sorry to anyone who did actual work on this but its my firm belief that this is a stupid change,entirely,stims were fine,pills were fine,its all un-necessary

**Comment by Al-1ce** [comment]
> I remember making a github account some time ago to protest against the removal of smartgunner from the game,since then there have been changes I have agreed with and disagreed with,but few I have disagreed with enough to say something,this is one of those times I'm saying something,terribly sorry to anyone who did actual work on this but its my firm belief that this is a stupid change,entirely,stims were fine,pills were fine,its all un-necessary

I've got to agree. Dislike/like ratio says enough.

**Comment by Araulius** [comment]
if you really want to nerf the pill, you should make the amount pills heal proportional to the medical skill of the user. I.e. if you are not a corpsman, your pills heal for less than an autoinjector of equivelent amount does.
